modules/tensor_mechanics/doc/content/modules/tensor_mechanics/2d_elastic_waves.md

```diff
@@ -0,0 +1,77 @@
+
+!row!
+!media Cantilever_beam.png
+      style=width:60%;
+      id=fig_cook
+      caption=2D cantilever problem with a prescribed displacement boundary condition on the right end.
+!row-end!
+
+The PDE that mathematically describes this problem is as follows:
+\begin{equation}
+    -\nabla .(E \hspace{1mm}\nabla u)+\rho \frac{\partial^2 u}{\partial t^2}=0 \label{eq1}
+\end{equation}
+
+with the following boundary conditions:
+\begin{equation}
+    u(x=0)=0, u(x=L)= a cos(2 \pi f t) \label{eq2}
+\end{equation}
+where $f$ is the frequency, $a=0.5$, and  $2 \pi f=\omega=10$.
+\\
+In equation 1, we consider Young's modulus and the density as constants and equal to unity.
+Equation (1) can be written as:
+\begin{equation}
+    -\nabla . (\nabla u)+\frac{1}{c_p^2} \frac{\partial^2 u}{\partial t^2} = 0 \label{eq3}
+\end{equation}
+where $c_p$ is the wave speed and $c_p=\sqrt{\frac{E}{\rho}}=1$ .
+
+Now, we consider $u=A e^{i(\omega t-kx)}$ is a solution for the equation (2) "hint: you can verify it by direct substitution", $\Longleftrightarrow$ $k^2= \frac{\omega^2}{c_p^2}=\omega^2$, where $k$ is the wave number.
+Now, equation (2) can be written as,
+\begin{equation}
+    -\nabla . (\nabla u)- k^2 u = 0 \label{eq4}
+\end{equation}
+Transforming equation (3) and the corresponding boundary condition to the frequency domain in time, we obtain:
+\begin{equation}
+    -\nabla . (\nabla U)- k^2 U = 0 \label{eq5}
+\end{equation}
+with the following boundary conditions:
+\begin{equation}
+    U(x=0)=0,  U(x=L)=0.5 \label{eq6}
+\end{equation}
+
+!alert note title= $U$ is a $\textbf{complex-valued}$ function
+At this stage, we split the complex sytem of equations into two $\textbf{real}$ systems, which yields two real-valued variables in each direction that are coupled and live on the same mesh (see the Kernels block).
+
+!listing examples/wave_propagation/2D_bar_elastic_wave_propagation.i block=Kernels
+
+!alert note
+$u$ is a function of time and space $(x,t)$, whereas $U$ is function of space and frequency $(x,\omega)$.
+
+The moose solution for this problem is shown below, see [fig_cook2].
+
+!row!
+!media wavefieldandprofile.png
+      style=width:100%;
+      id=fig_cook2
+      caption=the solution at $\omega =k =10$.
+!row-end!
+
+The analytical solution of equation (4) can be straight forward obtained knowing that we can express the solution of $U$ as a combination of sines and cosines representing the traveling and reflecting waves, i.e $U=C_1 e^{ikx} + C_2e^{-ikx}$.\\ \\
+Applying the boundary conditions, we obtain, $C_1=-C_2$ with $C_1=0.5/(e^{ikL}-e^{-ikL})$. So, the analytical solution is: $U=C_1(e^{ikx}-e^{-ikx}).$
+
+This can be simplified to be:
+\begin{equation}
+U= 0.5 \times [sin(kx)/sin(kL)] \label{eq7}
+\end{equation}
+
+# Input File Example
+
+The following block shows how you can define the kernels
+
+!listing examples/wave_propagation/2D_bar_elastic_wave_propagation.i
```

Comment:
I don't see how the imaginary and real parts are coupled in the input file.  Could you have just modeled the real part of this and completely ignored the imaginary part and gotten the same answer?